pkg/nexus/nexus.go

```diff
@@ -162,98 +179,64 @@ func (n *NexusDownloader) DownloadArtifact(c *config.MavenGav) (Deliverable, err
+	} else {
+		return fmt.Sprintf("%s-%s%s", c.ArtifactId, c.Version, getClassifierExt(c))
 	}
-	return resourceURL, nil
+}
 
+func createMavenManifestPath(c *config.MavenGav) string {
+	groupId := strings.ReplaceAll(c.GroupId, ".", "/")
+	return fmt.Sprintf("/repository/maven-intern/%s/%s/%s/maven-metadata.xml", groupId, c.ArtifactId, c.Version)
 }
 
-func (n *NexusDownloader) createURL(gav *config.MavenGav) (string, error) {
-	tmpURL, err := url.Parse(n.baseURL)
-	if err != nil {
-		return "", errors.Wrapf(err, "Failed to parse url")
+func createDownloadPath(manifest MavenManifest, c *config.MavenGav) string {
+	groupId := strings.ReplaceAll(c.GroupId, ".", "/")
+	versionWithoutSnapshot := strings.ReplaceAll(c.Version, "SNAPSHOT", "")
+
+	var artifact string
+	if c.IsSnapshot() {
+		artifact = fmt.Sprintf("%s-%s%s-%d%s", c.ArtifactId, versionWithoutSnapshot, manifest.Versioning.Snapshot.Timestamp, manifest.Versioning.Snapshot.BuildNumber, getClassifierExt(c))
+	} else {
+		artifact = fmt.Sprintf("%s-%s%s", c.ArtifactId, c.Version, getClassifierExt(c))
 	}
-	query := tmpURL.Query()
-	query.Set("g", gav.GroupId)
-	query.Set("a", gav.ArtifactId)
-	query.Set("v", gav.Version)
-	query.Set("e", string(gav.Type))
-	query.Set("c", string(gav.Classifier))
-	query.Set("r", "public-with-staging")
-	tmpURL.RawQuery = query.Encode()
-	return tmpURL.String(), nil
+	return fmt.Sprintf("/repository/maven-intern/%s/%s/%s/%s", groupId, c.ArtifactId, c.Version, artifact)
 }
 
-func (n *NexusDownloader) createNexus3URL(gav *config.MavenGav) (string, error) {
-	tmpURL, err := url.Parse(n.baseURL + "/service/rest/v1/search/assets/download")
-	if err != nil {
-		return "", errors.Wrapf(err, "Failed to parse url")
-	}
-	query := tmpURL.Query()
-	query.Set("sort", "version")
-	query.Set("repository", "maven-intern")
-	query.Set("maven.groupId", gav.GroupId)
-	query.Set("maven.artifactId", gav.ArtifactId)
-	query.Set("maven.extension", string(gav.Type))
-	query.Set("maven.classifier", string(gav.Classifier))
-	if gav.IsSnapshot() {
-		query.Set("maven.baseVersion", gav.Version)
+func getClassifierExt(c *config.MavenGav) string {
+
+	if c.Classifier != "" {
+		return fmt.Sprintf("-%s.%s", c.Classifier, c.Type)
 	} else {
-		query.Set("version", gav.Version)
+		return fmt.Sprintf(".%s", c.Type)
 	}
-	tmpURL.RawQuery = query.Encode()
-	return tmpURL.String(), nil
 }
 
-func (n *NexusDownloader) fileName(cfg *config.MavenGav, contentDisposition string, location string) (string, error) {
-	if len(contentDisposition) > 0 {
-		_, params, err := mime.ParseMediaType(contentDisposition)
-		if err != nil {
-			return "", errors.Wrap(err, "Failed to parse content-disposition")
-		}
-		fName := params["filename"]
-		if fName != "" {
-			return fName, nil
-		}
+func (n *BinaryDownloader) DownloadArtifact(c *config.MavenGav) (Deliverable, error) {
```

Comment:
Noen grunn til at denne er i BinaryDownloader når resten er i MavenDownloader?
Jeg ser den egentlig ikke er endret, men bare flyttet på grunn av diff... 
Rart navn, forøvrig. Den downloader jo egentlig ikke noe, bare sjekker at filen fins.

Reply:
Det er den som blir brukt for binary builds